ams/optimizer/src/main/java/com/netease/arctic/optimizer/local/LocalOptimizer.java

```diff
@@ -12,6 +12,11 @@ public class LocalOptimizer {
   public static void main(String[] args) throws CmdLineException {
     OptimizerConfig optimizerConfig = new OptimizerConfig(args);
     Optimizer optimizer = new Optimizer(optimizerConfig);
+
+    // calculate optimizer memory allocation
+    long memorySize = Runtime.getRuntime().maxMemory() / 1024 / 1024;
+    optimizerConfig.setMemorySize((int) memorySize);
+
```

Comment:
I tested it locally.
<img width="1671" alt="image" src="https://github.com/NetEase/amoro/assets/103108928/e7c056b3-6d44-4381-8fe5-8a4c07aa11a7">
Although there are some differences compared to -Xmx, I think the results are acceptable. 👍

Reply:
Yes, my local test also has a little gap. And the gap is not big, it feels acceptable.